pkg/virt-api/webhooks/validating-webhook/admitters/flavor-admitter.go

```diff
@@ -0,0 +1,139 @@
+package admitters
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	flavorv1alpha1 "kubevirt.io/client-go/apis/flavor/v1alpha1"
+	"kubevirt.io/kubevirt/pkg/flavor"
+	webhookutils "kubevirt.io/kubevirt/pkg/util/webhooks"
+	validating_webhooks "kubevirt.io/kubevirt/pkg/util/webhooks/validating-webhooks"
+)
+
+type FlavorAdmitter struct{}
+
+var _ validating_webhooks.Admitter = &FlavorAdmitter{}
+
+func (f *FlavorAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	return admitFlavor(ar,
+		metav1.GroupVersionResource{
+			Group:    flavorv1alpha1.SchemeGroupVersion.Group,
+			Version:  flavorv1alpha1.SchemeGroupVersion.Version,
+			Resource: "virtualmachineflavors",
+		},
+		func(raw []byte) ([]flavorv1alpha1.VirtualMachineFlavorProfile, error) {
+			flavorObj := flavorv1alpha1.VirtualMachineFlavor{}
+			err := json.Unmarshal(ar.Request.Object.Raw, &flavorObj)
+			if err != nil {
+				return nil, err
+			}
+			return flavorObj.Profiles, nil
+		},
+	)
+}
+
+type ClusterFlavorAdmitter struct{}
+
+var _ validating_webhooks.Admitter = &ClusterFlavorAdmitter{}
+
+func (f *ClusterFlavorAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	return admitFlavor(ar,
+		metav1.GroupVersionResource{
+			Group:    flavorv1alpha1.SchemeGroupVersion.Group,
+			Version:  flavorv1alpha1.SchemeGroupVersion.Version,
+			Resource: "virtualmachineclusterflavors",
+		},
+		func(raw []byte) ([]flavorv1alpha1.VirtualMachineFlavorProfile, error) {
+			clusterFlavorObj := flavorv1alpha1.VirtualMachineClusterFlavor{}
+			err := json.Unmarshal(ar.Request.Object.Raw, &clusterFlavorObj)
+			if err != nil {
+				return nil, err
+			}
+			return clusterFlavorObj.Profiles, nil
+		},
+	)
+}
+
+type extractProfilesFunc = func([]byte) ([]flavorv1alpha1.VirtualMachineFlavorProfile, error)
```

Comment:
If I remove the `=`, will it not be a type definition, instead of type alias?
In this case I wanted an alias, because it is used only to shorten variable definitions.